pkg/artifactory/resource_artifactory_local_repository.go

```diff
@@ -196,6 +196,9 @@ func resourceLocalRepositoryRead(d *schema.ResourceData, m interface{}) error {
 	c := m.(*ArtClient).ArtOld
 
 	repo, resp, err := c.V1.Repositories.GetLocal(context.Background(), d.Id())
+	if resp != nil {
```

Comment:
The equality check should be `resp == nil` but check `err` first. If you check error first you may never need to check if `resp` is `nil`. If `err != nil` then whatever values being returned are likely to be zero values and even if they aren't they can't be trusted because an error occurred. It's safer to always check the error first, otherwise it gets masked by whatever is wrong with the other returned values.

It looks like this happens in several places, it's probably worth hunting them down and fixing them since you're trying to establish a solid base.